### PR TITLE
Metadata emulator wait for change support

### PR DESCRIPTION
**Pull request by evildour**
This also adds support for concurrent requests, which are necessary so one request can update data while another request is hanging waiting for changes.

### CODE REVIEW

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs [hunk 1]
```diff
@@ -115,5 +120,15 @@ public void Dispose()
 
             Directory.Delete(emulatorPath, recursive: true);
         }
+
+        public Task UpdateMetadata(string path, string data)
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: You could use an expression-bodied method here, should you wish.

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs [hunk 2]
```diff
@@ -64,7 +68,8 @@ protected MetadataFixtureBase()
             int port = ((IPEndPoint)temp.LocalEndpoint).Port;
             temp.Stop();
 
-            Environment.SetEnvironmentVariable(EmulatorEnvironmentVariable, $"localhost:{port}");
+            emulatorHost = $"localhost:{port}";
+            Environment.SetEnvironmentVariable(EmulatorEnvironmentVariable, emulatorHost);
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: That only affects this process, right?

Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: Yes, but this change has no effect other than to cache the environment variable value in a field as well so we don't need to query the variable later.

**Comment by evildour** [comment]
(Python LGTM received offline from @alex-martelli)

**Comment by evildour** [comment]
(rebased)

**Pull request merged**